aten/src/ATen/native/cuda/BatchLinearAlgebraLib.cpp

```diff
@@ -367,6 +367,109 @@ Tensor _inverse_helper_cuda_lib(const Tensor& self) {
   return self_inv_working_copy;
 }
 
+// call cusolver gesvd function to calculate svd
+template<typename scalar_t>
+inline static void _apply_svd_lib_gesvd(const Tensor& self, Tensor& U, Tensor& S, Tensor& VT, Tensor& infos, bool compute_uv, bool some,
+  const bool calculate_all_batches,
+  const std::vector<int64_t>& batches
+) {
+  using value_t = typename c10::scalar_value_type<scalar_t>::type;
+  auto self_data = self.data_ptr<scalar_t>();
+  auto U_data = U.data_ptr<scalar_t>();
+  auto S_data = S.data_ptr<value_t>();
+  auto VT_data = VT.data_ptr<scalar_t>();
+  auto self_stride = matrixStride(self);
+  auto U_stride = !compute_uv ? 1 : matrixStride(U);
+  auto S_stride = S.size(-1);
+  auto VT_stride = !compute_uv ? 1 : matrixStride(VT);
+
+  int _batchsize = cuda_int_cast(batchCount(self), "batch size");
+  int m = cuda_int_cast(self.size(-2), "m");
+  int n = cuda_int_cast(self.size(-1), "n");
+  int lda = std::max<int>(1, m);
+  int ldvt = std::max<int>(1, n);
+
+  TORCH_INTERNAL_ASSERT(m >= n, "cusolver gesvd only supports matrix with sizes m >= n");
+
+  char jobu = compute_uv ? (some ? 'S' : 'A') : 'N';
+  char jobvt = compute_uv ? (some ? 'S' : 'A') : 'N';
+  auto handle = at::cuda::getCurrentCUDASolverDnHandle();
+
+  int lwork = -1;
+  at::cuda::solver::gesvd_buffersize<scalar_t>(handle, m, n, &lwork);
+  TORCH_INTERNAL_ASSERT(lwork >= 0, "gesvd_buffersize failed to get needed buffer size, got lwork = ", lwork);
```

Comment:
`lwork=0` should be invalid. cuSOLVER doesn't specify the size of `work` array, but LAPACK documentation says it should be `max(1, lwork)` and I assume the behavior here would be the same as in LAPACK. I haven't tested whether using `lwork=0` actually raises an error from cuSOLVER for any input size.
```suggestion
  TORCH_INTERNAL_ASSERT(lwork >= 1, "gesvd_buffersize failed to get needed buffer size, got lwork = ", lwork);
```

Reply:
It's not written in the cusolver doc whether `lwork` would be `>= 0` or `1`, and it's not guaranteed to be the same as lapack. To make it future-proof, I tend to leave it as `>= 0`, or simply remove the check.